script.js

```diff
@@ -35,22 +35,22 @@ function startVideo() {
 startVideo();
 
 video.addEventListener('play', () => {
-  
+
   setInterval(async () => {
-    const detections = await faceapi.detectAllFaces(video, 
+    const detections = await faceapi.detectAllFaces(video,
       new faceapi.TinyFaceDetectorOptions())
       .withFaceExpressions()
-    if (detections[0].expressions.happy > 0.3)
+    if (detections && detections[0] && detections[0].expressions.happy > 0.3)
       if (scorePlayer)
-        happyFacedetected();  
+        happyFacedetected();
   }, 1000)
 });
 
 function happyFacedetected() {
   let currentScore = parseInt(document.getElementById('score').innerText);
   if (currentScore == 0) {
-    snackNotif(firstLaugh[parseInt(Math.random() * firstLaugh.length)]);  
-  }  
+    snackNotif(firstLaugh[parseInt(Math.random() * firstLaugh.length)]);
+  }
   else
     snackNotif(Laugh[parseInt(Math.random() * Laugh.length)]);
```

Comment:
please save `parseInt(Math.random() * firstLaugh.lenght` in a var we should do this operation only once may be put it in a separate function maybe and use bot in the if...else blocks ?

Reply:
firstLaugh.length will have to passed to that function then, I think its a simple operation and we don't repeat it often so this looks fine to me

Reply:
it should be fine but should keep the logic at one place